### ISSUE TITLE
Pivot Row %/Column % are exactly halved when Show Margins is on (rows total 50%)

### ISSUE THREAD

**Issue by AshishK714** [social-engineer-ai/PyAnalytica#6]
**Labels:** bug, severity:high, explore, correctness

## Environment

| | |
|---|---|
| PyAnalytica | 0.7.0 (installed from `dist/pyanalytica-0.7.0-py3-none-any.whl` into a clean venv) |
| Python | 3.14 |
| Key deps | shiny 1.7.0, pandas 3.0.5, numpy 2.5.2, scipy 1.18.1, matplotlib 3.11.1, seaborn 0.13.2 |
| OS / browser | Windows 11, Chrome |
| Dataset | `bank_campaign.csv` — 41,188 rows x 16 cols (UCI Bank Marketing, columns renamed) |

Found during a systematic 153-case sweep of every panel against one dataset.
Numbers were checked against an independent pandas/scipy/sklearn computation, not just
"did it render".

## Summary

In Explore > Pivot, `Normalize = Row %` (or `Column %`) with **Show Margins** ticked divides
each row by a sum that includes the `All` margin, so every value comes out at half its true
size and each row totals 50% instead of 100%.

## Steps to reproduce

1. Load `bank_campaign.csv`.
2. Explore > **Pivot**: Row = `channel`, Column = `subscribed`, Value = `age`,
   Aggregation = `count`, **Show Margins on**, Normalize = **Row %**.
3. Click Create Pivot.

## Expected vs actual

| Row | Expected No / Yes | Actual No / Yes |
|---|---|---|
| Landline | 94.77 / 5.23 | **47.4 / 2.6** |
| Mobile | 85.26 / 14.74 | **42.6 / 7.4** |
| Row total | 100 | **50** |

Reproduced on a second, unrelated pair (`age_band` x `subscribed`): 30-39 reads 44.9 / 5.1
instead of 89.9 / 10.1. Every value is exactly half; the error is independent of the variables.

## Cause

`pyanalytica/explore/pivot.py`:

```python
result = pd.pivot_table(df, index=index, columns=columns, values=values,
                        aggfunc=aggfunc, margins=margins, observed=True)

if normalize and aggfunc == "count":
    if normalize == "index":
        row_sums = result.sum(axis=1)          # <-- includes the "All" column
        result = result.div(row_sums, axis=0) * 100
    elif normalize == "columns":
        col_sums = result.sum(axis=0)          # <-- includes the "All" row
        result = result.div(col_sums, axis=1) * 100
    elif normalize == "all":
        total = result.iloc[:-1, :-1].sum().sum() if margins else result.sum().sum()  # correct
```

The `all` branch already excludes the margins with `iloc[:-1, :-1]`; the `index` and `columns`
branches do not.

## Not affected

- Explore > **Cross-tab** — delegates to `pd.crosstab`, which normalises before adding margins,
  and returns the correct 94.8 / 5.2.
- Pivot with **Show Margins off** — correct.

## Suggested fix

Exclude the margin from the divisor in the same way the `all` branch does:

```python
if normalize == "index":
    base = result.iloc[:, :-1] if margins else result
    result = result.div(base.sum(axis=1), axis=0) * 100
```

or delegate to `pd.crosstab` for the count+normalize case.

## Evidence

`shots/B17_DEFECT_margins_rowpct.jpg`

---

*Found in a systematic 153-case sweep of every panel against a 41,188-row teaching dataset, with each number checked against an independent pandas/scipy/scikit-learn computation rather than against the app itself.*

**Comment by AshishK714** [comment]
Fixed in 0.7.1. The divisor now excludes the margin row/column, and Show Code prints the slice that actually ran. Note that `test_pivot_normalize_index` was passing *because* of this bug -- it summed the `All` column into the row, so 50 + 50 also came to 100.
